Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -295,3 +400,97 @@ extension SecureEnclaveValet {
     }
 
 }
+
+// MARK: - Private Extensions
+
+extension SecureEnclaveAccessControl {
+
+    /// The `LAPolicy` that the `SecureEnclaveAccessControl` corresponds to.
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    fileprivate var policy: LAPolicy {
+        switch self {
+        case .userPresence, .devicePasscode:
+            return .deviceOwnerAuthentication
+        case .biometricAny, .biometricCurrentSet:
+            if #available(macOSApplicationExtension 10.12.2, *) {
+                return .deviceOwnerAuthenticationWithBiometrics
+            } else {
+                return .deviceOwnerAuthentication
+            }
+        }
+    }
+    #endif
+
+}
+
+// MARK: - Internal Types
+
+/// A type that statically takes an `LAError` and a `SecureEnclaveAccessControl`
+/// and transforms it to either a `KeychainError` or `SecureEnclaveError`
+enum LAErrorTransformer {
+
+    enum TransformedError: Equatable {
+        case keychain(KeychainError)
+        case secureEnclave(SecureEnclaveError)
+    }
+
+    #if !os(watchOS)
+    @available(tvOS 10.0, *)
+    static func transform(
+        error: LAError,
+        accessControl: SecureEnclaveAccessControl
+    ) -> TransformedError {
+        switch (error.code, accessControl) {
+        case (.userCancel, _),
+             (.systemCancel, _),
+             (.authenticationFailed, _),
+             (.appCancel, _):
+            return .secureEnclave(.userCancelled)
+
+        case (.userFallback, _):
+            return .secureEnclave(.userFallback)
+
+        case (.touchIDLockout, _),
+             (.invalidContext, _),
+             (.notInteractive, _),
```

Comment:
Is this an `.internalError`? Seems like [this error is a result of misconfiguring a `LAContext`](https://developer.apple.com/documentation/localauthentication/laerror/2873509-notinteractive?changes=latest_major)

Reply:
yeah, sounds more like an `internalError`, or maybe a `configurationError` if we're defining `internalError` as things that should never happen

Reply:
is there a case where we'd reasonably hit the `notInteractive` case? From the docs, it seems like we'd have to set the [`interactionNotAllowed`](https://developer.apple.com/documentation/localauthentication/lacontext/2873508-interactionnotallowed?changes=latest_major) property ourselves on the `LAContext` to hit this? Maybe I'm missing something though!

Reply:
Good point! I wasn't thinking about how we are in full control of the `LAContext`. I feel more comfortable making that and `invalidContext` `internalError`s in that case